### PR TITLE
tools: add cgroup driver patcher (to enable systemd driver)

### PR DESCRIPTION
**Pull request by sk4zuzu**

### CODE REVIEW

File: tools/development/k8s/memory/patch_cgroup_driver/ansible.cfg [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+retry_files_enabled=false
+any_errors_fatal=true
+
+[privilege_escalation]
+become=true
+become_user=root
+become_method=sudo
+
+[ssh_connection]
+ssh_pipelining=true
```
Comment (to-bar) [inline tools/development/k8s/memory/patch_cgroup_driver/ansible.cfg]:
```suggestion
ssh_pipelining=false
```

File: tools/development/k8s/memory/patch_cgroup_driver/apply.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+set -e
+
+read -p "This script will stop and reconfigure kubelet/docker services, then reboot all k8s worker nodes. Are you sure (y/N)? " CHOICE
+case "$CHOICE" in
+  y|yes|Y|Yes) ;;
+  *) exit ;;
+esac
+
+exec ansible-playbook -vv main.yml
```
Comment (to-bar) [inline tools/development/k8s/memory/patch_cgroup_driver/apply.sh]:
```suggestion
ansible-playbook -vv main.yml
```
I think it's better to see in process tree what called `ansible-playbook`

File: tools/development/k8s/memory/patch_cgroup_driver/readme.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+## Usage
+
+1. Copy your cluster's `inventory` file to this directory.
+
+2. Run `sh apply.sh`.
+
+## Procedure
+
+Ansible will sequentially (rolling update) reconfigure `docker` and `kubelet` services on each worker node.
```
Comment (to-bar) [inline tools/development/k8s/memory/patch_cgroup_driver/readme.md]:
```suggestion
Ansible will sequentially (rolling update but **without waiting for pods to be `Ready`**) reconfigure `docker` and `kubelet` services on each worker node.
```

File: tools/development/k8s/memory/patch_cgroup_driver/readme.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+When there are any changes found in config files, ansible will (for each worker node):
+
+1. Drain all user-deployed pods.
+
+2. Stop docker and kubelet services.
+
+3. Reconfigure docker and kubelet services with the systemd cgroup setting.
```
Comment (to-bar) [inline tools/development/k8s/memory/patch_cgroup_driver/readme.md]:
```suggestion
3. Reconfigure docker and kubelet services to use `systemd` cgroup driver.
```

File: tools/development/k8s/memory/patch_cgroup_driver/main.yml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  serial: 1
+  environment:
+    KUBECONFIG: /etc/kubernetes/admin.conf
+  tasks:
+    - name: Define cgroup drivers
+      set_fact:
+        old_cgroup_driver: >-
+          {{ old_cgroup_driver | default('cgroupfs') }}
+        new_cgroup_driver: >-
+          {{ new_cgroup_driver | default('systemd') }}
```
Comment (to-bar) [inline tools/development/k8s/memory/patch_cgroup_driver/main.yml]:
```suggestion
      set_fact:
        old_cgroup_driver: cgroupfs
        new_cgroup_driver: systemd
```

**Review by to-bar — CHANGES_REQUESTED** [review]

File: tools/development/k8s/memory/patch_cgroup_driver/main.yml (hunk 5 above)
Comment (sk4zuzu) [reply tools/development/k8s/memory/patch_cgroup_driver/main.yml]: I moved this to `vars` and added the `rollback` procedure! :+1::innocent:

**Review by to-bar — APPROVED** [review]

**Review by erzetpe — APPROVED** [review]

**Pull request merged**